tests/unittests/tests-credman/tests-credman.c

```diff
@@ -0,0 +1,160 @@
+
+    /* add invalid credential params */
+    credential.params.psk = NULL;
+    ret = credman_add_credential(&credential);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_ERROR, ret);
+    TEST_ASSERT_EQUAL_INT(exp_count, credman_get_used_count());
+
+    /* fill the system credential buffer */
+    credential.params.psk = &exp_psk_params;
+    while (credman_get_used_count() < CREDMAN_MAX_CREDENTIALS) {
+        /* increase tag number so that it is not recognized as duplicate */
+        credential.tag++;
+        TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_add_credential(&credential));
+        TEST_ASSERT_EQUAL_INT(++exp_count, credman_get_used_count());
+    }
+
+    /* add to full system credential buffer */
+    credential.tag++;
+    ret = credman_add_credential(&credential);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NO_SPACE, ret);
+    TEST_ASSERT_EQUAL_INT(exp_count, credman_get_used_count());
+}
+
+static void test_credman_get_credential(void)
+{
+    int ret;
+    credman_credential_t out_credential;
+    credman_credential_t exp_credential = {
+        .tag = CREDMAN_TEST_TAG,
+        .type = CREDMAN_TYPE_ECDSA,
+        .params = { .ecdsa = &exp_ecdsa_params }
+    };
+
+    /* get non-existing credential */
+    ret = credman_get_credential(&out_credential, exp_credential.tag,
+                                 exp_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NOT_FOUND, ret);
+
+    ret = credman_add_credential(&exp_credential);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+
+    ret = credman_get_credential(&out_credential, exp_credential.tag,
+                                 exp_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(&out_credential, &exp_credential,
+                                 sizeof(exp_credential)));
+}
+
+static void test_credman_delete_credential(void)
+{
+    int ret;
+    unsigned exp_count = 0;
+    credman_credential_t exp_credential = {
+        .tag = CREDMAN_TEST_TAG,
+        .type = CREDMAN_TYPE_ECDSA,
+        .params = { .ecdsa = &exp_ecdsa_params }
+    };
+
+    /* delete non-existing credential */
+    ret = credman_delete_credential(exp_credential.tag, exp_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NOT_FOUND, ret);
+    TEST_ASSERT_EQUAL_INT(exp_count, credman_get_used_count());
+
+    /* add a credential */
+    ret = credman_add_credential(&exp_credential);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+    TEST_ASSERT_EQUAL_INT(++exp_count, credman_get_used_count());
+
+    /* delete a credential from system buffer */
+    ret = credman_delete_credential(exp_credential.tag, exp_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+    TEST_ASSERT_EQUAL_INT(--exp_count, credman_get_used_count());
+
+    /* delete a deleted credential */
+    ret = credman_delete_credential(exp_credential.tag, exp_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NOT_FOUND, ret);
+    TEST_ASSERT_EQUAL_INT(exp_count, credman_get_used_count());
```

Comment:
A test case for 

```C
credman_add_credential(&credential1);
credman_add_credential(&credential2);
credman_add_credential(&credential3);
credman_delete_credential(&credential1);
credman_delete_credential(&credential2);
credman_delete_credential(&credential3);
```

should exist ;-)

Reply:
Added new test case.